### PR TITLE
Load/save presets

### PR DESCRIPTION
**Pull request by BadiBasso**
#16 First version. Known bug which needs to be solved: when tool used first time and a network is added list is populated twice:
sl capture

After first load of preset this error disapperars and adding/removing networks work as intended.
Notices:
-last list of networks is not yet loaded/saved at start/exit. Edit: This feature is now included in PR
-only additional networks are saved/loaded, not the selected 'main' Network
-UI strings are not yet localized

### CODE REVIEW

File: ParallelRoadTool/NetTypeItem.cs [hunk 1]
```diff
@@ -15,4 +15,24 @@ public NetTypeItem(NetInfo netInfo, float horizontalOffset, float verticalOffset
...
+    public class PresetNetItem
+    {
+        public float HorizontalOffset;
+        public bool IsReversed;
+        public string NetName;
+        public float VerticalOffset;
+
+        public PresetNetItem(string netName, float horizontalOffset, float verticalOffset, bool isReversed)
+        {
+            NetName = netName;
```
Comment (ST-Apps) [inline ParallelRoadTool/NetTypeItem.cs]: Do you have the chance to change your game's language and see if you can still load a preset saved while using English? I'm not 100% sure but I think that we can't use network's name as it depends on game's language.

File: ParallelRoadTool/UI/UIUtils.cs [hunk 2]
```diff
@@ -0,0 +1,203 @@
+﻿using System.Collections.Generic;
```
Comment (ST-Apps) [inline ParallelRoadTool/UI/UIUtils.cs]: I think we already have some of these things in the UIUtil class, consider merging them to keep only one.

File: ParallelRoadTool/UI/UIMainWindow.cs [hunk 3]
```diff
@@ -22,14 +22,18 @@ public class UIMainWindow : UIPanel
             new SavedInt("toggleY", ParallelRoadTool.SettingsFileName, -1000, true);
 
         private UIOptionsPanel _mainPanel;
-        private UINetList _netList;
+        //private UINetList _netList;
+        public UINetList _netList;
```
Comment (ST-Apps) [inline ParallelRoadTool/UI/UIMainWindow.cs]: I'd keep this private and add utility methods to access it.

File: ParallelRoadTool/NetTypeItem.cs (hunk 1 above)
Comment (BadiBasso) [reply ParallelRoadTool/NetTypeItem.cs]: Tested with German. Seems to work just fine.  Names were saved in English to XML even if language was set to German.
English saved preset, loaded in German: OK
German saved preset, loaded in English: OK

Same preset loaded in English and in German:
sl english

sl german

File: ParallelRoadTool/UI/UIMainWindow.cs (hunk 3 above)
Comment (BadiBasso) [reply ParallelRoadTool/UI/UIMainWindow.cs]: Yeah, this is part of the import hack that I don't like.

File: ParallelRoadTool/UI/UIUtils.cs (hunk 2 above)
Comment (BadiBasso) [reply ParallelRoadTool/UI/UIUtils.cs]: Will do. This first version was a quick port from MoveIt just to get it work.

**Comment by BadiBasso** [comment]
Saved networks get loaded at start but they do not show in the list before an additional network is added (which still appear as double :( )

**Comment by ST-Apps** [comment]
Merging into ISSUE-16 branch so that I can check it out and test it, thanks.

**Comment by BadiBasso** [comment]
Rewrote this for refactored codebase; import code is not a hack anymore and there are no problems at start. Also got rid of the additional UIUtils class. New PR coming soon.

**Comment by ST-Apps** [comment]
> there are no problems at start

That's why I decided to do a refactoring: we had duplicated data structures and I suspected that was causing the issues you described.
Thanks for your help :)

**Pull request merged**